src/cargo_ops/elaborate_workspace.rs

```diff
@@ -301,4 +320,100 @@ impl<'ela> ElaborateWorkspace<'ela> {
+            let depth = path.len() as i32 - 1;
+            // generate lines
+            let status = &self.pkg_status.borrow_mut()[&path];
+            if (status.compat.is_changed() || status.latest.is_changed())
+                && (options.flag_packages.is_empty()
+                    || options.flag_packages.contains(&pkg.name().to_string()))
+            {
+                // name version compatible latest kind platform
+                let parent = path.get(path.len() - 2);
+
+                let mut line;
+
+                if let Some(parent) = parent {
+                    let dependency = &self.pkg_deps[parent][pkg];
+                    let label = if self.workspace_mode
+                        || parent == &self.workspace.current()?.package_id()
+                    {
+                        pkg.name().to_string()
+                    } else {
+                        format!("{}->{}", self.pkgs[parent].name(), pkg.name())
+                    };
+
+                    let dependency_type = match dependency.kind() {
+                        Kind::Normal => "Normal",
+                        Kind::Development => "Development",
+                        Kind::Build => "Build",
+                    };
+
+                    line = Metadata {
+                        name: label,
+                        project: pkg.version().to_string(),
+                        compat: status.compat.to_string(),
+                        latest: status.latest.to_string(),
+                        kind: Some(dependency_type.to_string()),
+                        platform: dependency.platform().map(|p| p.to_string()),
+                    };
+                } else {
+                    line = Metadata {
+                        name: pkg.name().to_string(),
+                        project: pkg.version().to_string(),
+                        compat: status.compat.to_string(),
+                        latest: status.latest.to_string(),
+                        kind: None,
+                        platform: None,
+                    };
+                }
+
+                crate_graph.dependencies.insert(line);
+            }
+            // next layer
+            if options.flag_depth.is_none() || depth < options.flag_depth.unwrap() {
+                self.pkg_deps[pkg]
+                    .keys()
+                    .filter(|dep| !path.contains(dep))
+                    .filter(|dep| {
+                        !self.workspace_mode
+                            || !self.workspace.members().any(|mem| &mem.package_id() == dep)
+                    })
+                    .for_each(|dep| {
+                        let mut path = path.clone();
+                        path.push(dep);
+                        queue.push_back(path);
+                    });
+            }
+        }
+
+        if crate_graph.dependencies.is_empty() {
+            if !self.workspace_mode {
+                println!("All dependencies are up to date, yay!");
```

Comment:
That will break JSON

Reply:
Ooops! That was pulled in from `print_list()`! Resolved.